### PR TITLE
[openrct2] Add types for OpenRCT2 plugin framework

### PR DESCRIPTION
**Pull request by IntelOrca**
This adds [openrct2.d.ts](https://github.com/OpenRCT2/OpenRCT2/blob/develop/distribution/openrct2.d.ts) that is distributed with [OpenRCT2](https://github.com/OpenRCT2/OpenRCT2) to aid writing of plugins.

OpenRCT2 uses [duktape](https://duktape.org) to run JavaScript plugins. Plugins written in either JavaScript, or TypeScript can currently reference `openrct2.d.ts` from the game using `/// <reference...`. This now allows them to omit that, and instead just have a devDependency on `@types/openrct2`.

**Examples:**
* [Title Sequence Editor](https://github.com/OpenRCT2/title-sequence-editor)
* [DevTools](https://github.com/OpenRCT2/plugin-devtools)
* [Advanced Track](https://github.com/oli414/AdvancedTrack)
* [Ride Vehicle Editor](https://github.com/Basssiiie/OpenRCT2-RideVehicleEditor)
* More: http://openrct2plugins.org

Because the types already exist in the OpenRCT2 repository, and are updated there when new APIs are introduced, I have kept the version in here 1:1 identical. For this reason I have disabled the necessary tslint rules that prevented the tests from passing, in the hope that this package can be exempt from that restriction.

**Checks:**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
@IntelOrca is that expected to be somehow recommended approach by the OpenRCT2 community? You folks have documented using the inline imports in source code and on your project wikis. My only personal concerns are a discoverability of the work and a consistent experience. Should we open an issue on the project's repo?
(note: I've seen you're the original implementation owner of the project's plugin system in JS/TS)

**Comment by IntelOrca** [comment]
@peterblazejewicz since it was implemented, an inline import was the simplest way to get some decent auto completion and docs when writing a plugin in JavaScript in vscode. Very simple to set up for people that haven't ever touched npm before as well. Over the past year though, a lot of plugins have been written using npm, babel and TypeScript, and have GitHub repositories. These would benefit from obtaining the type definitions from a package, so that CI systems can build the plugins. If this is merged, I will update the plugin documentation with information about importing the types by npm.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@IntelOrca thx!

File: types/openrct2/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,2456 @@
...
+         * are shared with `setInterval`.
+         * @param handle The numerical handle of the registered timeout to remove.
+         */
+        clearTimeout(handle: number): void;
+    }
+
+    interface Configuration {
+        getAll(namespace: string): { [name: string]: any };
+        get<T>(key: string): T | undefined;
+        get<T>(key: string, defaultValue: T): T;
```
Comment (DanielRosenwasser) [inline types/openrct2/index.d.ts:275]: I do think these might be better off as `unknown`; would that be acceptable for most users?

Comment (IntelOrca) [reply types/openrct2/index.d.ts:275]: If plugins are already using this, e.g. `let x = get<number>("language", 0);`, would that plugin then fail to build if the definition was changed upstream?

**Comment by elibarzilay** [comment]
@IntelOrca: is this ready to merge?

**Comment by IntelOrca** [comment]
@elibarzilay yep

**Pull request merged**